tests/test_extract.py

```diff
@@ -0,0 +1,159 @@
+    oi_1305 = fits.Column('OI_1304', format='1E', unit='count /s', disp='G15.6', array=np.random.rand(10, 1))
+    oi_1356 = fits.Column('OI_1356', format='1E', unit='count /s', disp='G15.6', array=np.random.rand(10, 1))
+    darkrate = fits.Column('DARKRATE', format='1E', unit='count /s /pixel', disp='G15.6', array=np.random.rand(10, 1))
+    col_defs = fits.ColDefs(
+        [time, longitude, latitude, sun_alt, sun_zd, target_alt, radial_vel, shift1, ly_alpha, oi_1305, oi_1356,
+         darkrate])
+    time_hdu = fits.BinTableHDU.from_columns(col_defs)
+    time_hdu.name = "TIMELINE"
+    # create an hdu list
+    hdu_list = fits.HDUList([prim_hdu, hdu, gti_hdu, time_hdu])
+    # write the hdu to a fits file
+    if os.path.exists(file):
+        os.remove(file)
+    hdu_list.writeto(file)
+    hdu_list = fits.open(file)
+    return hdu_list
+
+
+# generate_fits_file("test.fits")
+
+def test_get_columns():
+    """
+    Test if the function is returning the right column fields
+    """
+    # Setup
+    test_data = generate_fits_file("Output/lbgu17qnq_corrtag_a.fits")
+    dt = test_data[1].data
+    detector = "FUV"
+    # detector = "NUV"
+
+    # Truth actual values
+    # Testing for FUV
+    xfull = dt.field("xfull")
+    # xfull = dt.field("xdopp") not necessary for current detector
+    if cosutil.findColumn(dt, "yfull"):
+        yfull = dt.field("yfull")
+    else:
+        yfull = dt.field("ycorr")
+    dq = dt.field("dq")
+    epsilon = dt.field("epsilon")
+
+    # Test function
+    (xf, yf, dq2, epsilon2) = extract.getColumns(test_data, detector)
+
+    # Verify
+    np.testing.assert_array_equal(xfull, xf)
+    np.testing.assert_array_equal(yfull, yf)
+    np.testing.assert_array_equal(dq, dq2)
+    np.testing.assert_array_equal(epsilon, epsilon2)
+
+
+def test_remove_unwanted_column():
+    """
+    Old column length should be equal to new column length + amount of the removed columns
+    """
+    # Setup
+    target_cols = ['XFULL', 'YFULL']
+    # Truth
+    fd = generate_fits_file("Output/lbgu17qnq_lampflash.fits")
+    table = fd[1].data
+    cols = table.columns
+
+    # Test
+    fd = extract.remove_unwanted_columns(fd)
+    new_cols = fd[1].data.columns
+    # Verify
+    deleted_cols = set(cols) - set(new_cols)
+    deleted_cols = np.array(list(deleted_cols))
+    temp_cols = [d.name for d in deleted_cols]
+    deleted_cols = deleted_cols[np.argsort(temp_cols)]
+    assert target_cols[0] == deleted_cols[0].name
+    assert target_cols[1] == deleted_cols[1].name
+
+
+def test_next_power_of_two():
+    """
+    Test the next_power_of_two
+    @return: none
+    """
+    # Truth
+    next_power = 8
+
+    # Verify
+    assert next_power == extract.next_power_of_two(7)
+
+
+# todo *****************************
+
+
+def test_add_column_comment():
+    # Setup
+    ofd = generate_fits_file("Output/myFitsFile.fits")
+    comment = "Time in seconds"
```

Comment:
It might be better to change this comment to something obscure that wouldn't be in the generate_fits_file function by default to test as the truth